Userland/Libraries/LibGfx/ImageFormats/JPEGWriter.cpp

```diff
@@ -136,20 +136,36 @@ class JPEGEncodingContext {
         return {};
     }
 
+    static Array<double, 106> create_cosine_lookup_table()
+    {
+        static constexpr double pi_over_16 = AK::Pi<double> / 16;
+
+        Array<double, 106> table;
+        table.fill(NAN);
+
+        for (u8 u = 0; u < 8; ++u) {
+            for (u8 x = 0; x < 8; ++x)
+                table[(2 * x + 1) * u] = cos((2 * x + 1) * u * pi_over_16);
```

Comment:
Instead of leaving half the entries NaN, couldn't you index this by `u * 8 + x` so you use fewer entries and calculations?

Reply:
Yep, that works too